### PR TITLE
Publish association

### PR DESCRIPTION
**Pull request by BlueWaves2**
## Description of changes
Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

  
## How has this been tested?
Please describe the test strategy you followed.

- [X] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [X] manual, exploratory test

Manual Test was performed over compiling the course-, content-, and media service. Over graphiql a course and a chapter was created. With the chapter ID a content was created. And over the Content ID created a new media entity.
    
## Checklist before requesting a review
- [x] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code fulfills all acceptance criteria
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [x] I have made corresponding changes to the documentation
- [x] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [x] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/media_service/dapr/TopicPublisher.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import io.dapr.client.DaprClientBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+/**
+ * Component that takes care of publishing messages to a dapr Topic
+ */
+@Component
+@Slf4j
+public class TopicPublisher {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/media_service/dapr/TopicPublisher.java:16]: Stupid question but will this be moved in the common code in the future? would make sense, right?

File: src/main/java/de/unistuttgart/iste/gits/media_service/service/MediaService.java [hunk 2]
```diff
@@ -22,6 +24,10 @@
 @Slf4j
 public class MediaService {
 
+    public static final String BUCKET_ID = "bucketId";
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/media_service/service/MediaService.java:27]: No please, that's horrible :D If you really want a constant, it would "Media record with id {} not found" and you to string formatting

Comment (PaulBredl) [reply src/main/java/de/unistuttgart/iste/gits/media_service/service/MediaService.java:27]: Sorry I  meant the NOT_FOUND and MEDIA_RECORD_WITH_ID

File: src/main/java/de/unistuttgart/iste/gits/media_service/service/MediaService.java [hunk 3]
```diff
@@ -226,16 +246,16 @@ private Map<String, String> createMinIOVariables(UUID input) {
         Map<String, String> variables = new HashMap<>();
 
         if (!repository.existsById(input)) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/media_service/service/MediaService.java:248]: could be extracted in common method

**Review by PaulBredl — APPROVED** [review]
small issues but nothing for today
